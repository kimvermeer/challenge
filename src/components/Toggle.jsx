// @flow
import React from "react";

type Props = {
  isChecked: boolean,
  onChange: () => void,
  value: string
};

class Toggle extends React.Component {
  static props: Props;

  state = {
    isChecked: false
  };

  componentWillReceiveProps(nextProps) {
    if (this.state.isChecked !== nextProps.isChecked) {
      this.setState({
        isChecked: nextProps.isChecked
      });
    }
  }

  onClick = () => {
    this.setState({ isChecked: !this.state.isChecked });
    this.props.onChange(this.props.value);
  };

  render() {
    return (
      <div className="toggle-wrapper">
        <input
          type="checkbox"
          className="toggle hidden"
          name={this.props.value}
          checked={this.state.isChecked}
          onChange={this.onClick}
          value={this.props.value}
        />
        <label htmlFor={this.props.value} onClick={this.onClick} />
        <span>{this.props.value}</span>
      </div>
    );
  }
}

export default Toggle;
