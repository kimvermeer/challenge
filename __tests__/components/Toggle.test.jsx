import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Toggle from "../../src/components/Toggle";

const defaultProps = {
  value: "B1",
  onChange: jest.fn()
};

describe("Toggle", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Toggle {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("checks the input when clicking on the label", () => {
    const wrapper = mount(<Toggle {...defaultProps} />);

    expect(wrapper.state("isChecked")).toBeFalsy();
    wrapper.find("label").simulate("click");
    expect(wrapper.state("isChecked")).toBeTruthy();
  });
});
