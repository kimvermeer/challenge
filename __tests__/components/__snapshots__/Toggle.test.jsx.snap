// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly 1`] = `
<div
  className="toggle-wrapper"
>
  <input
    checked={false}
    className="toggle hidden"
    name="B1"
    onChange={[Function]}
    type="checkbox"
    value="B1"
  />
  <label
    htmlFor="B1"
    onClick={[Function]}
  />
  <span>
    B1
  </span>
</div>
`;
